src/Features/VisualBasic/Portable/ConvertLinq/VisualBasicConvertLinqQueryToLinqMethodProvider.vb

```diff
@@ -0,0 +1,176 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.Composition
+Imports Microsoft.CodeAnalysis
+Imports Microsoft.CodeAnalysis.CodeRefactorings
+Imports Microsoft.CodeAnalysis.ConvertLinq
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+Imports Microsoft.CodeAnalysis.LanguageServices
+Imports Microsoft.CodeAnalysis.VisualBasic.Utilities
+Imports System.Threading
+Imports Microsoft.CodeAnalysis.VisualBasic
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.ConvertLinq
+    <ExportCodeRefactoringProvider(LanguageNames.VisualBasic, Name:=NameOf(VisualBasicConvertLinqQueryToLinqMethodProvider)), [Shared]>
+    Partial Friend NotInheritable Class VisualBasicConvertLinqQueryToLinqMethodProvider
+        Inherits AbstractConvertLinqProvider
+
+        Protected Overrides Function CreateAnalyzer(syntaxFacts As ISyntaxFactsService, semanticModel As SemanticModel) As IAnalyzer
+            Return New VisualBasicAnalyzer(syntaxFacts, semanticModel)
+        End Function
+
+
+        Private NotInheritable Class VisualBasicAnalyzer
+            Inherits Analyzer(Of QueryExpressionSyntax, ExpressionSyntax)
+            Public Sub New(syntaxFacts As ISyntaxFactsService, semanticModel As SemanticModel)
+                MyBase.New(syntaxFacts, semanticModel)
+            End Sub
+
+            Protected Overrides ReadOnly Property Title() As String
+                Get
+                    Return VBFeaturesResources.Convert_linq_query_to_linq_method
```

Comment:
logic is the same for C#.  Introduce common base type and pull this up to there.

Reply:
I'm afraid that naming will be different for some features in Linq Convert. For example, there will be different foreach and For Each conversions. Therefore, we will not be able to share all the names. Do you think it is still reasonable to share some names and have separate ones ones?

---
In reply to: [167387956](https://github.com/dotnet/roslyn/pull/24759#discussion_r167387956) [](ancestors = 167387956)